### PR TITLE
Streamline and fix CI. 

### PR DESCRIPTION
**Pull request by jhodgdev**

### CODE REVIEW

**Comment by jhodgdev** [comment]
This branch is for altering our CI workflow so that `hlint` and `fourmolu` are both installed via nix rather than Slack. This will hopefully prevent tooling mismatches meaning we get inconsistent results when using the `Makefile`.

**Comment by jhodgdev** [comment]
And it might potentially save us some GitHub minutes and therefore $$$

**Comment by jhodgdev** [comment]
I have altered the CI workflow so that hlint and fourmolu now install via Nix rather than Stack. This has fixed the CI issues we were getting wherein mismatched tool versions were in some cases preventing `make format` and `make lint` from detecting issues that the CI would find and complain about. CI times are now as quick as the Stack solution (when the tools were cached successfully) and significantly quicker than the Stack solution when the tools were _not_ cached.

File: apropos-tx.cabal [hunk 1]
```diff
@@ -3,6 +3,7 @@ name:               apropos-tx
 version:            1.0
 extra-source-files: CHANGELOG.md
 
+-- triggering CI AGAIN
```
Comment (delicious-lemon) [inline apropos-tx.cabal]: What's this?

**Review by delicious-lemon — APPROVED** [review]
Awesome! :+1:

File: apropos-tx.cabal (hunk 1 above)
Comment (jhodgdev) [reply apropos-tx.cabal]: Nothing you saw... :/

**Pull request merged**